vllm/model_executor/parallel_utils/communication_op.py

```diff
@@ -104,3 +109,67 @@ def broadcast_object_list(obj_list, src=0):
     # Broadcast.
     torch.distributed.broadcast_object_list(obj_list, src=src)
     return obj_list
+
+
+class TensorMetadata:
+    """A simple class to hold tensor metadata."""
+
+    def __init__(self, tensor):
+        self.dtype = tensor.dtype
+        self.size = tensor.size()
+
+    def __repr__(self):
+        return (f"TensorMetadata(dtype={self.dtype}, size={self.size})")
+
+
+def broadcast_tensor_dict(tensor_dict: Dict[Any, Union[torch.Tensor,
+                                                       Any]] = None,
```

Comment:
The `tensor_dict` type is a bit weird: Shouldn't it be `Optional[Dict]`? Also, how `Union[torch.Tensor, Any]` is different from `Any`?

Reply:
Thanks I indeed missed an `Optional`. I use `Union[torch.Tensor, Any]` to explicitly emphasize that we treat `torch.Tensor` and other types differently in this function.